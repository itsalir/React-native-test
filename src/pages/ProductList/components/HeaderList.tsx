import {StyleSheet, Text, TextInput, View} from 'react-native';
import React from 'react';
import Icon from 'react-native-vector-icons/Ionicons';
import {ProductListType} from '../../../api/types';
import {SafeAreaView} from 'react-native-safe-area-context';

type Props = {
  data?: ProductListType[];
  onChangeText: (text: string) => void;
};

const HeaderList = ({data, onChangeText}: Props) => {
  return (
    <SafeAreaView>
      <View style={styles.container}>
        <Icon name="menu-outline" size={30} />
        <View style={styles.searchSection}>
          <Icon
            style={styles.searchIcon}
            name="search"
            size={20}
            color="#000"
          />
          <TextInput
            style={styles.input}
            placeholder="Search product"
            onChangeText={onChangeText}
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
      <View style={styles.container}>
        <View style={styles.containerView}>
          <Text style={styles.title}>Sneakers</Text>
          <Text style={styles.subTitle}>
            {data?.length || 0} products found
          </Text>
        </View>
        <Icon name="funnel-outline" size={24} />
      </View>
    </SafeAreaView>
  );
};

export default HeaderList;

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#F4F4F4',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    margin: 10,
    gap: 20,
    zIndex: 999,
  },
  title: {
    color: '#000',
    fontSize: 30,
    fontStyle: 'normal',
    fontWeight: '800',
  },
  subTitle: {
    color: '#000',
    fontSize: 16,
    fontStyle: 'normal',
    fontWeight: '400',
  },
  containerView: {
    gap: 10,
  },
  searchSection: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#fff',
    borderRadius: 20,
  },
  searchIcon: {
    padding: 10,
  },
  input: {
    flex: 1,
    paddingTop: 10,
    paddingRight: 10,
    paddingBottom: 10,
    paddingLeft: 0,
    backgroundColor: '#fff',
    color: '#424242',
  },
});
