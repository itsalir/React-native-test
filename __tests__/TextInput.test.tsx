import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import HeaderList from '../src/pages/ProductList/components/HeaderList';

test('renders the HeaderList component with a TextInput', () => {
  const onChangeTextMock = jest.fn();
  const {getByPlaceholderText} = render(
    <HeaderList data={[]} onChangeText={onChangeTextMock} />,
  );

  const searchInput = getByPlaceholderText('Search product');
  fireEvent.changeText(searchInput, 'Sample Text');

  expect(onChangeTextMock).toHaveBeenCalledWith('Sample Text');
});

it('minimum props', () => {
  const onChangeTextMock = jest.fn();
  const {toJSON} = render(
    <HeaderList data={[]} onChangeText={onChangeTextMock} />,
  );

  expect(toJSON()).toMatchSnapshot();
});
