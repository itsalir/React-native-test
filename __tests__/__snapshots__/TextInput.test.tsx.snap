// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`minimum props 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F4F4F4",
        "flexDirection": "row",
        "gap": 20,
        "justifyContent": "space-between",
        "margin": 10,
        "zIndex": 999,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 30,
          },
          undefined,
          {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 20,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#000",
              "fontSize": 20,
            },
            {
              "padding": 10,
            },
            {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
      <TextInput
        onChangeText={[MockFunction]}
        placeholder="Search product"
        style={
          {
            "backgroundColor": "#fff",
            "color": "#424242",
            "flex": 1,
            "paddingBottom": 10,
            "paddingLeft": 0,
            "paddingRight": 10,
            "paddingTop": 10,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F4F4F4",
        "flexDirection": "row",
        "gap": 20,
        "justifyContent": "space-between",
        "margin": 10,
        "zIndex": 999,
      }
    }
  >
    <View
      style={
        {
          "gap": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000",
            "fontSize": 30,
            "fontStyle": "normal",
            "fontWeight": "800",
          }
        }
      >
        Sneakers
      </Text>
      <Text
        style={
          {
            "color": "#000",
            "fontSize": 16,
            "fontStyle": "normal",
            "fontWeight": "400",
          }
        }
      >
        0
         products found
      </Text>
    </View>
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 24,
          },
          undefined,
          {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
  </View>
</RNCSafeAreaView>
`;
